helpers/scrolling.ts

```diff
@@ -139,18 +145,52 @@ export function scrollHorizontally(
   if (left - parentLeft < opts.leftOffset) {
     const goal = scroll.x - opts.leftOffset + left - parentLeft;
     if (opts.smooth) {
-      smoothScrollLeft(goal, parent);
+      addToScrollQueue(smoothScrollLeft, parent, goal);
     } else {
-      scrollElement(parent, goal, scroll.y);
+      addToScrollQueue(scrollElement, parent, goal, scroll.y);
     }
   }
 
   if (right - parentLeft > width - opts.rightOffset) {
     const goal = scroll.x + right - parentLeft - width + opts.rightOffset;
     if (opts.smooth) {
-      smoothScrollLeft(goal, parent);
+      addToScrollQueue(smoothScrollLeft, parent, goal);
     } else {
-      scrollElement(parent, goal, scroll.y);
+      addToScrollQueue(scrollElement, parent, goal, scroll.y);
     }
   }
 }
+
+type ScrollFunction = (element: Element | Window, x: number, y?: number) => Promise<void>;
+
+type ScrollQueueItem = {
+  element: Element | Window;
+  fn: ScrollFunction;
+  x: number;
+  y?: number;
+};
+
+const queue: ScrollQueueItem[] = [];
+let queueRunning: boolean;
+
+function addToScrollQueue(
+  fn: ScrollFunction,
+  element: Element | Window,
+  x: number,
+  y?: number
+): void {
+  queue.push({ fn, element, x, y });
+  if (!queueRunning) {
+    processQueue();
+  }
+}
+
+function processQueue() {
+  if (queue.length > 0) {
+    queueRunning = true;
+    const { fn, element, x, y } = queue.splice(0, 1)[0];
+    fn(element, x, y).finally(processQueue);
```

Comment:
Do we really support `finally` ? Did you add polyfill for it in SQ build ?

Reply:
We do, we added a polyfill for it. However, now I'm wondering if we need the polyfill for the SUC build as well 🤔 ... We shouldn't, right?

Reply:
Ok thanks, I'll add a polyfill for it in SC too... Even with the drop of legacy browsers it seems we still support browsers that don't have `finally`.

I guess as soon as we want to make it public we will need to have a cleaner build indeed, but for now we don't have any polyfill at all and the build is very basic (just doing a `tsc`). So it rely on the fact that SQ and SC have the polyfill...

Reply:
I've been investigating that polyfill thing and the current situation is a bit weird.
For SQ the modern build has no polyfill at all and would not need any for `finally` since it's supposed to support the last 2 browsers versions.
The legacy build has polyfill and support IE11 and last 3 browsers version.
BUT the switch between legacy and modern build is done based on the [es module support of the browser](https://caniuse.com/#feat=es6-module) which mean there is a lot of browsers version that might end up with the modern build without being fully supported by it.
And all browsers that get the legacy build except IE11 are not fully supported by the build either, since last 3 browsers are all compatible with es modules and will get the modern build...

And we have a simpler but similar mess with SC...

Reply:
@wouter-admiraal-sonarsource Since it looks like the situation needs to be clearer, could you NOT use `finally` here for now ? @gregaubert and I are planning a task force to take the power back on this matter (at least to try :laughing: ).